src/test/kotlin/com/personio/synthetics/builder/api/StepBuilderTest.kt

```diff
@@ -146,6 +150,43 @@ class StepBuilderTest {
         assertEquals(isCritical, result.isCritical)
     }
 
+    @Test
+    fun `build sets step retry defaults if retry method was not called`() {
+        val stepBuilder = StepBuilder("any_name", makeRequestBuilderHappyPathMock())
+        stepBuilder.request { }
+        val result = stepBuilder.build()
+
+        assertEquals(SyntheticsTestOptionsRetry(), result.retry)
+    }
+
+    @Test
+    fun `build sets step retry values if retry method was called`() {
+        val stepBuilder = StepBuilder("any_name", makeRequestBuilderHappyPathMock())
+        stepBuilder.request { }
+        stepBuilder.retry(3, 3.seconds)
+        val result = stepBuilder.build()
+
+        assertEquals(SyntheticsTestOptionsRetry().count(3).interval(3000.0), result.retry)
+    }
+
+    @ParameterizedTest
```

Comment:
It would be cleaner if you check one thing at a time, like [here](https://github.com/personio/datadog-synthetic-test-support/blob/master/src/test/kotlin/com/personio/synthetics/config/BrowserTestConfigTest.kt#L98C7-L98C7) and [here](https://github.com/personio/datadog-synthetic-test-support/blob/master/src/test/kotlin/com/personio/synthetics/config/BrowserTestConfigTest.kt#L105)